log/value.go

```diff
@@ -28,23 +30,31 @@ func Value(ctx context.Context, v interface{}) interface{} {
 	return v
 }
 
-// Caller returns returns a Valuer that returns a pkg/file:line description of the caller.
+// Caller returns a Valuer that returns a pkg/file:line description of the caller.
 func Caller(depth int) Valuer {
-	return func(context.Context) interface{} {
-		_, file, line, _ := runtime.Caller(depth)
-		if strings.LastIndex(file, "/log/filter.go") > 0 {
-			depth++
-			_, file, line, _ = runtime.Caller(depth)
-		}
-		if strings.LastIndex(file, "/log/helper.go") > 0 {
-			depth++
-			_, file, line, _ = runtime.Caller(depth)
-		}
+	return func(ctx context.Context) interface{} {
+		curDepth := getSkipDepth(ctx)
+		_, file, line, _ := runtime.Caller(depth + curDepth + defaultDepth)
```

Comment:
Why the `defaultDepth` should be add to each Caller depth? `default` should mean it value could be overrided, it may lead to some misunderstandings.

Reply:
I've resubmitted the code, will this be easier to understand?